src/app/page.tsx

```diff
@@ -169,15 +137,17 @@ export default function Chat() {
                   <input
                     className="flex-auto pl-3 h-12 mr-5 pr-28 py-2 bg-transparent placeholder:text-slate-400 text-slate-400 text-sm transition duration-300 ease focus:outline-none focus:border-slate-400 hover:border-slate-400 shadow-sm focus:shadow-md"
                     placeholder="Specify a topic you want the joke to be about."
+                    name="topic"
+                    value={state.topic}
+                    onChange={handleChange}
                   />
-                  {/* Generate Joke button */}
                   <button
                     className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded disabled:opacity-50"
-                    disabled={isLoading || !state.genre || !state.tone}
+                    disabled={isLoading || !state.genre || !state.tone || !state.topic}
                     onClick={() =>
                       append({
                         role: "user",
-                        content: `Generate a ${state.genre} story in a ${state.tone} tone`,
+                        content: `Generate a ${state.genre} joke in a ${state.tone} tone about ${state.topic}`,
```

Comment:
I can see this change basically implemented `topic` in the message sent to the backend AI. I have checked `main` branch and this change is already implemented in line 110 `sendRequestToOpenAI`

Reply:
https://github.com/LuisJoseSanchez/encode-club-ai-and-gpt-bootcamp-q3-homework-week2/blob/935b9388e747eb4afd66ef56831bc4ba26ab31fe/src/app/page.tsx#L110